flumeserver/indexer/indexer.go

```diff
@@ -159,22 +157,19 @@ func ProcessDataFeed(feed datafeed.DataFeed, csConsumer transports.Consumer, txF
 	pruneTicker := time.NewTicker(5 * time.Second)
 	txCount := 0
 	txDedup := make(map[common.Hash]struct{})
-	defer sub.Unsubscribe()
 	defer txSub.Unsubscribe()
 	db.Exec("DELETE FROM mempool.transactions WHERE 1;")
 	for {
 		select {
 		case <-quit:
 			if !processed {
-				panic("Shutting down without processing any blocks")
-				//try not to panic, pass back into main and do proper cleanup
+					fmt.Errorf("Shutting down without processing any blocks")
```

Comment:
The purpose of this panic was to exit with a non-zero status code so that the snapshotting process could identify when it had failed to take a successful snapshot.

Ideally, we'd have a channel or something where we could notify the main channel and have it do an `os.Exit(1)`, but just returning here means that it will shutdown cleanly without any way to detect that we didn't sync anything.

Reply:
Could I os.Exit(1) from this function?